### ISSUE TITLE
Refactor EvidenceLink Note field to be locally controlled field. - [merged]

### ISSUE THREAD

**Issue by benloh** [theRAPTLab/meme#96]
_Merges dev-bl/evlink-note-refactor -> dev-bl/ui-tweaks_

The EvidenceLink "Note" field was a controlled field.  Text input changes were sent to PMCData which in turn Published a `DATA_CHANGED` event.  The parent EvidenceList component Subscribed to the `DATA_CHANGED` event and would update the `note` prop for the EvidenceLinks it listed.  On the EvidenceLink itself the note field was handled via the prop passed by EvidenceList.

We refactored EvidenceLink (and EvidenceList) to make it a locally controlled field.  Rather than updating the note field via props, we now update the field via local state variable.  Any input field changes result in:
1. setting this.state.note to match the new field value
2. sending the updated field to PMCData to update the data store.
In other words, the component handles local updates and passes on data changes to the data manager.

The initial value of the note field is still read via props.

# To Test

From a user perspective, the functionality should be exactly the same as it was before:

1. Expand a Resource Item to view the EvidenceLinks related to the resource.
2. Expand an EvidenceLink to view the Edit button.
3. Click "Edit"
4. Type in the field.  The field label should update as you type.  If the controlled component was not working, your typing would result in no changes (or perhasp flawed changes).
5. Click "Save"
6. Click on the ResourceItem itself to open the ResourceView
7. Review the EvidenceLink -- the text should reflect your changes.

**Comment by benloh** [comment]
mentioned in commit 507f39557e67400373138274984132fdc57264e9

**Comment by benloh** [comment]
merged
